internal/agent/server.go

```diff
@@ -46,7 +64,15 @@ func (s *Server) Start(statusUpdater *service.StatusUpdater) {
 
 	s.restServer = &http.Server{Addr: fmt.Sprintf("0.0.0.0:%d", s.port), Handler: router}
 
-	// Run the server
+	if s.tlsConfig != nil {
+		zap.S().Named("server").Info("tls configured")
+		s.restServer.TLSConfig = s.tlsConfig
+		if err := s.restServer.ListenAndServeTLS("", ""); err != nil && err != http.ErrServerClosed {
+			zap.S().Named("server").Fatalf("failed to start server with tls: %w", err)
+		}
+	}
+
+	// Run the server without tls
 	err := s.restServer.ListenAndServe()
```

Comment:
Won't this crash as we try to run http+https on same port?

Reply:
`ListenAndServe` is a blocking op. it's either one or the other